openhands/llm/llm.py

```diff
@@ -179,6 +181,8 @@ def attempt_on_error(retry_state):
 
         def custom_completion_wait(retry_state):
             """Custom wait function for litellm completion."""
+            if should_exit():
+                raise UserCancelledError('Shutdown requested during retry by user.')
```

Comment:
Could you please help me understand why does this happen now, what breaks it. Why do we need to add a custom mechanism if we didn't before?

Reply:
This has been moved to the `before_sleep` handler, which is called _before_ `custom_completion_wait` to end the loop.